module/motion/QuinticWalk/src/engine/TrajectoryUtils.hpp

```diff
@@ -0,0 +1,307 @@
+                return value >= other.value;
+            }
+            bool operator==(TrajectoryTypes const& other) const {
+                return value == other.value;
+            }
+            bool operator!=(TrajectoryTypes const& other) const {
+                return value != other.value;
+            }
+            bool operator<(TrajectoryTypes::Value const& other) const {
+                return value < other;
+            }
+            bool operator>(TrajectoryTypes::Value const& other) const {
+                return value > other;
+            }
+            bool operator<=(TrajectoryTypes::Value const& other) const {
+                return value <= other;
+            }
+            bool operator>=(TrajectoryTypes::Value const& other) const {
+                return value >= other;
+            }
+            bool operator==(TrajectoryTypes::Value const& other) const {
+                return value == other;
+            }
+            bool operator!=(TrajectoryTypes::Value const& other) const {
+                return value != other;
+            }
+
+            // Conversions
+            operator Value() const {
+                return value;
+            }
+            operator uint8_t() const {
+                return value;
+            }
+
+            operator std::string() const {
+                switch (value) {
+                    case Value::IS_DOUBLE_SUPPORT: return "IS_DOUBLE_SUPPORT";
+                    case Value::IS_LEFT_SUPPORT_FOOT: return "IS_LEFT_SUPPORT_FOOT";
+                    case Value::TRUNK_POS_X: return "TRUNK_POS_X";
+                    case Value::TRUNK_POS_Y: return "TRUNK_POS_Y";
+                    case Value::TRUNK_POS_Z: return "TRUNK_POS_Z";
+                    case Value::TRUNK_AXIS_X: return "TRUNK_AXIS_X";
+                    case Value::TRUNK_AXIS_Y: return "TRUNK_AXIS_Y";
+                    case Value::TRUNK_AXIS_Z: return "TRUNK_AXIS_Z";
+                    case Value::FOOT_POS_X: return "FOOT_POS_X";
+                    case Value::FOOT_POS_Y: return "FOOT_POS_Y";
+                    case Value::FOOT_POS_Z: return "FOOT_POS_Z";
+                    case Value::FOOT_AXIS_X: return "FOOT_AXIS_X";
+                    case Value::FOOT_AXIS_Y: return "FOOT_AXIS_Y";
+                    case Value::FOOT_AXIS_Z: return "FOOT_AXIS_Z";
+                    default:
+                        throw std::runtime_error("enum TrajectoryTypes's value is corrupt, unknown value stored"
+                                                 + std::to_string(static_cast<uint8_t>(value)));
+                }
+            }
+        };
+
+        /**
+         * Simple typedef for trajectories container
+         */
+        using Trajectories = SplineContainer<SmoothSpline<float>, TrajectoryTypes, float>;
+
+        /**
+         * Return initialized trajectories for trunk/foot ik cartesian with empty splines
+         */
+        inline Trajectories TrajectoriesInit() {
```

Comment:
@Bidski @TrentHouliston if you have time can you have a quick look at this function and see if you can find a reason for memory issues?